backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/backup/StartVmBackupCommand.java

```diff
@@ -264,6 +268,24 @@ protected void executeCommand() {
         Guid vmBackupId = createVmBackup();
         log.info("Created VmBackup entity '{}'", vmBackupId);
 
+        // Set a VDS to be able to gather Qemu Image Info
+        if (getVds() == null) {
+            setHostForColdBackupOperation();
```

Comment:
here, can't we try an choose a host that has this capability?
using something like this:
https://github.com/oVirt/ovirt-engine/blob/193cfccd8ef706dd2e704763eae235c42c2f1578/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/backup/StartVmBackupCommand.java#L427

but with a complex predicate that also checks if if bitmap info is supported

Reply:
We could do that indeed.
But are there a lot of cases where 1 host in the cluster does support this feature and another not?

And neither there is a helper for getHostForExecution that you can supply multiple Predicates.
In fact we should call

setHostForQemuImgInfo -> which sets the VDS to a host that preferably supports isColdBackupEnabled and QemuImgInfo
If that is not available, one that only supports QemuImgInfo.

and then in setHostForColdBackupOperation we should check if the host set has support for isColdBackupEnabled, and otherwise try to find another one.

Reply:
@dupondje I apologize for missing this, feel free to ping in an additional comment if this happens again

>And neither there is a helper for getHostForExecution that you can supply multiple Predicates.

What I meant is something like this:

```java
Predicate<VDS> coldBackupEnabled = VDS::isBackupEnabled;
Predicate<VDS> supportsBitmapInfo = VDS::isQemuImageInfoBitmaps;
getParameters().setVdsRunningOn(
                    vdsCommandsHelper.getHostForExecution(getStoragePoolId(), coldBackupEnabled.and(supportsBitmapInfo)));
```